### PR TITLE
Quality of life updates to conditions

### PR DESCRIPTION
**Pull request by piotr-roslaniec**
**Type of PR:**
- Feature

**Required reviews:**
- 2

**What this does:**
- Adds `or` and `and` shorthand methods to `CompoundCondition`
- Removes mandatory `conditionType` from conditions constructors. It's now inferred inside the constructor. The condition props still contain the `conditionType` field to ensure it's validation.

**Issues fixed/closed:**
> - Fixes #...

**Why it's needed:**
> Explain how this PR fits in the greater context of the NuCypher Network.
> E.g., if this PR address a `nucypher/productdev` issue, let reviewers know!

**Notes for reviewers:**
- Based on https://github.com/nucypher/nucypher-ts/pull/365, please review it first

### CODE REVIEW

File: packages/taco/src/conditions/base/index.ts [hunk 1]
```diff
@@ -1,37 +1,70 @@
...
+  constructor(value: OmitType<CompoundConditionProps>) {
+    super(compoundConditionSchema, {
+      conditionType: CompoundConditionType,
+      ...value,
+    });
+  }
+
+  private static withOperator(
+    operands: ConditionProps[],
+    operator: 'or' | 'and',
```
Comment (derekpierre) [inline packages/taco/src/conditions/base/index.ts:27]: Is there already an issue filed to deal with `not` compound conditions in 7.0.0? This is completed on the `nucypher` side -https://github.com/nucypher/nucypher/pull/3293

File: packages/taco/src/conditions/base/index.ts [hunk 2]
```diff
@@ -1,37 +1,70 @@
...
+import { Condition, ConditionProps } from '../condition';
 
-import { ContractConditionProps, contractConditionSchema } from './contract';
-import { RpcConditionProps, rpcConditionSchema } from './rpc';
-import { TimeConditionProps, timeConditionSchema } from './time';
+import { ContractConditionProps, contractConditionSchema, ContractConditionType } from './contract';
+import { RpcConditionProps, rpcConditionSchema, RpcConditionType } from './rpc';
+import { TimeConditionProps, timeConditionSchema, TimeConditionType } from './time';
+
+type OmitType<T> = Omit<T, 'conditionType'>
```
Comment (derekpierre) [inline packages/taco/src/conditions/base/index.ts:12]: `OmitType<...>` reads a little odd for me, like "omit the type that's being provided to it" almost eg. OmitType<Foo> almost reads like "omit the Foo type" (to me).

What do you think about `OmitConditionType<...>` or `OmitConditionTypeFrom<...>`? or something like that?

File: packages/taco/test/conditions/base/contract.test.ts [hunk 3]
```diff
@@ -43,6 +43,11 @@ describe('validation', () => {
       },
     });
   });
+
+  it('infers condition type from constructor', ()=> {
+    const condition = new ContractCondition(testContractConditionObj);
+    expect(condition.value.conditionType).toEqual(ContractConditionType);
+  })
```
Comment (derekpierre) [inline packages/taco/test/conditions/base/contract.test.ts:50]: Seems like `testContractConditionObj` does contain `conditionType` in it, does it also make sense to additionally test removing the value from `testContractConditionObj` and then passing it in? This ensures that the Omit functionality is ok if the `conditionType` isn't provided. Does that make sense or would that situation never happen? If it makes sense, similar comment for other analogous tests that infer type.

File: packages/taco/test/conditions/compound-condition.test.ts [hunk 4]
```diff
@@ -166,4 +165,36 @@ describe('validation', () => {
     expect(result.error).toBeDefined();
     expect(result.data).toBeUndefined();
   });
+
+  it('accepts shorthand for "or" operator', () => {
```
Comment (derekpierre) [inline packages/taco/test/conditions/compound-condition.test.ts:169]: Nice test.

File: packages/taco/src/conditions/base/index.ts (hunk 1 above)
Comment (piotr-roslaniec) [reply packages/taco/src/conditions/base/index.ts:27]: Filed an issue here: https://github.com/nucypher/nucypher-ts/issues/369

File: packages/taco/src/conditions/base/index.ts (hunk 2 above)
Comment (piotr-roslaniec) [reply packages/taco/src/conditions/base/index.ts:12]: I thought it was a bit wordy, but it make sense

**Review by derekpierre — APPROVED** [review]
🎸 - looks good.

There is a failing CI job.

File: examples/taco/nextjs/src/app/page.tsx [hunk 5]
```diff
@@ -73,16 +73,15 @@ function App() {
 
     console.log('Encrypting message...');
     const hasPositiveBalance = new conditions.RpcCondition({
-      conditionType: 'rpc',
-      chain: 5,
+      chain: 80001,
```
Comment (derekpierre) [inline examples/taco/nextjs/src/app/page.tsx:76]: @piotr-roslaniec / @KPrasch Does changing the chain from 5 -> 80001 affect any default behaviour / expectations of the demo...?

Comment (piotr-roslaniec) [reply examples/taco/nextjs/src/app/page.tsx:76]: It doesn't but it makes the example more consistent (everything is happening on the same chain).

Comment (KPrasch) [reply examples/taco/nextjs/src/app/page.tsx:76]: The tapir nodes are not evaluating chain 5 for conditions, but 80001 is used for both testnet and devnet.  I figure this gives the examples/demos the best shot at working "out of the box" regardless of which testnet you use.

**Review by KPrasch — APPROVED** [review]
